apps/admin/components/google-signin.tsx

```diff
@@ -0,0 +1,110 @@
+"use client";
+
+import { FC, useEffect, useRef, useCallback, useState } from "react";
+import Script from "next/script";
+import axios, { AxiosError } from "axios";
+import { GOOGLE_SIGN_IN_URL } from "@/api-routes";
+import {
+  TGoogleSignInErrorResponse,
+  TGoogleSignInSuccessResponse,
+} from "@/types";
+import { cookieSetter } from "@/actions";
+import { useRouter } from "next/navigation";
+
+type Props = {};
+
+interface GsiButtonConfiguration {
+  type: "standard" | "icon";
+  theme?: "outline" | "filled_blue" | "filled_black";
+  size?: "large" | "medium" | "small";
+  text?: "signin_with" | "signup_with" | "continue_with" | "signup_with";
+  shape?: "rectangular" | "pill" | "circle" | "square";
+  logo_alignment?: "left" | "center";
+  width?: number;
+  local?: string;
+}
+
+export const GoogleSignInButton: FC<Props> = () => {
+  const router = useRouter();
+  const clientId = process.env.NEXT_PUBLIC_GOOGLE_CLIENT_ID;
+  const handleGoogleSignIn = async ({ clientId, credential }: any) => {
+    if (!clientId || !credential) {
+      throw Error("Cant find credentials");
+    }
+    try {
+      const response = await axios.post<TGoogleSignInSuccessResponse>(
+        GOOGLE_SIGN_IN_URL,
+        {
+          auth_token: credential,
+        }
+      );
+      cookieSetter({
+        access: response.data.tokens.access,
+        refresh: response.data.tokens.refresh,
+      });
+      router.push("/admin/dashboard");
+    } catch (err) {
+      const error = err as AxiosError<TGoogleSignInErrorResponse>;
+      console.error(error);
```

Comment:
```suggestion
      console.error(error);
```
is the above console required?